spec/matchers/be_called_with_arguments_spec.rb

```diff
@@ -26,6 +27,37 @@
       it 'correctly matches when wildcard is used for all arguments' do
         expect(@command).to be_called_with_arguments(anything, anything)
       end
+
+      it 'displays the diff between what was called and what was expected' do
+        begin
+          expect(@command).to be_called_with_arguments('not_first_argument', 'second_argument')
+        rescue RSpec::Expectations::ExpectationNotMetError => rex
+          error_string = <<-multiline_string
+expected stubbed_command to be called with arguments ["not_first_argument", "second_argument"]
+Diff:
+@@ -1,2 +1,3 @@
+-["not_first_argument", "second_argument"]
++["first_argument", "second_argument"]
++["third_argument", "fourth_argument"]
+multiline_string
+          expect(rex.message.uncolorize).to eql error_string
+        end
+      end
+
+      it 'displays the diff between what was called and what was expected (negative case)' do
+        begin
+          expect(@command).to_not be_called_with_arguments('first_argument', 'second_argument')
+        rescue RSpec::Expectations::ExpectationNotMetError => rex
+          expected_error_string = <<-multiline_string
+expected stubbed_command not to be called with arguments ["first_argument", "second_argument"]
+Diff:
+@@ -1,2 +1,3 @@
+ ["first_argument", "second_argument"]
```

Comment:
The diff format here is a good compromise, in my opinion, as it shows what did and didn't match (doesn't handle anything very well, but it's a start).